core/primitives/src/utils.rs

```diff
@@ -202,56 +208,38 @@ impl KeyForAccessKey {
         }
         PublicKey::try_from_slice(&raw_key[prefix_len..])
     }
-}
-
-#[derive(derive_more::AsRef, derive_more::Into)]
-#[as_ref(forward)]
-pub struct KeyForData(Vec<u8>);
-
-impl TrieKey for KeyForData {}
-
-impl KeyForData {
-    pub fn estimate_len(account_id: &AccountId, data: &[u8]) -> usize {
-        KeyForAccount::estimate_len(&account_id) + ACCOUNT_DATA_SEPARATOR.len() + data.len()
-    }
-
-    pub fn get_prefix_with_capacity(account_id: &AccountId, reserved_capacity: usize) -> Self {
-        let mut prefix: Vec<u8> = KeyForAccount::with_capacity(
-            account_id,
-            ACCOUNT_DATA_SEPARATOR.len() + reserved_capacity,
-        )
-        .into();
-        prefix.extend(ACCOUNT_DATA_SEPARATOR);
-        Self(prefix)
-    }
 
-    pub fn get_prefix(account_id: &AccountId) -> Self {
-        Self::get_prefix_with_capacity(account_id, 0)
-    }
-
-    pub fn with_suffix(&self, suffix: &[u8]) -> Self {
-        let mut raw_key = Vec::with_capacity(self.0.len() + suffix.len());
-        raw_key.extend(&self.0);
-        raw_key.extend(suffix);
-        Self(raw_key)
-    }
-
-    pub fn new(account_id: &AccountId, data: &[u8]) -> Self {
-        let mut key = Self::get_prefix_with_capacity(&account_id, data.len());
-        key.0.extend(data);
-        debug_assert_eq!(key.0.len(), Self::estimate_len(&account_id, &data));
-        key
+    pub fn parse_data_key_from_contract_data_key<'a>(
+        raw_key: &'a [u8],
+        account_id: &AccountId,
+    ) -> Result<&'a [u8], std::io::Error> {
```

Comment:
should we have customized error instead of coercing into `std::io::Error`? Is it then converted to `StorageError` in runtime?

Reply:
It's used in many places. Usually, it's just unwrapped with `expect`